llama_stack/providers/inline/post_training/torchtune/common/checkpointer.py

```diff
@@ -81,83 +91,175 @@ def save_checkpoint(
-        if source_path.exists():
-            shutil.copy(
-                source_path,
-                Path.joinpath(model_file_path, "params.json"),
-            )
-        source_path = Path.joinpath(self._checkpoint_dir, "tokenizer.model")
-        if source_path.exists():
-            shutil.copy(
-                source_path,
-                Path.joinpath(model_file_path, "tokenizer.model"),
-            )
-        source_path = Path.joinpath(self._checkpoint_dir, "orig_params.json")
-        if source_path.exists():
-            shutil.copy(
-                source_path,
-                Path.joinpath(model_file_path, "orig_params.json"),
+            if not adapter_only:
+                model_state_dict = state_dict[training.MODEL_KEY]
+                if self._model_type == ModelType.LLAMA3_VISION:
+                    from torchtune.models.llama3_2_vision._convert_weights import (
+                        llama3_vision_tune_to_meta,
+                    )
+
+                    state_dict[training.MODEL_KEY] = llama3_vision_tune_to_meta(
+                        model_state_dict
+                    )
+                else:
+                    # llama3_2 has tied weights, so we need to add the output.weight key
+                    if (
+                        self._model_type == ModelType.LLAMA3_2
+                        and "output.weight" not in model_state_dict
+                    ):
+                        model_state_dict["output.weight"] = model_state_dict[
+                            "tok_embeddings.weight"
+                        ]
+
+                    state_dict[training.MODEL_KEY] = convert_weights.tune_to_meta(
+                        model_state_dict
+                    )
+
+                model_file_name = Path.joinpath(model_file_path, "consolidated.00.pth")
+
+                torch.save(state_dict[training.MODEL_KEY], model_file_name)
+                logger.info(
+                    "Model checkpoint of size "
+                    f"{os.path.getsize(model_file_name) / 1000**3:.2f} GB "
+                    f"saved to {model_file_name}"
+                )
+
+            if training.ADAPTER_KEY in state_dict:
+                adapter_file_path = model_file_path / "adapter"
+                adapter_file_path.mkdir(parents=True, exist_ok=True)
+                adapter_file_name = Path.joinpath(adapter_file_path, "adapter.pth")
+                torch.save(state_dict[training.ADAPTER_KEY], adapter_file_name)
+                logger.info(
+                    "Adapter checkpoint of size "
+                    f"{os.path.getsize(adapter_file_name) / 1000**3:.2f} GB "
+                    f"saved to {adapter_file_name}"
+                )
+
+            elif adapter_only:
+                raise ValueError(
+                    "Adapter checkpoint not found in state_dict. Please ensure that the state_dict contains adapter weights."
+                )
+        elif checkpoint_format == "hf":
```

Comment:
This code will be more readable to by making separate methods for each format instead of inlining all code in one place